src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -198,11 +198,23 @@ public ModelAndView inputSanitizationNonCompliant(@RequestParam String favoriteC
     }
     
     public void processResponseNoncompliant(Map<String, String> response) {
-    // Noncompliant: logs sensitive data without redaction resulting in a possible sensitive information leak.
-    final String name = response.get("Name");
-    String redacted = RedactionUtil.redact(name);
-    log.info(name);
-}
+        // Noncompliant: logs sensitive data without redaction resulting in a possible sensitive information leak.
+        final String name = response.get("Name");
+        String redacted = RedactionUtil.redact(name);
+        log.info(name);
+    }
+    
+    public void executeSqlStatementNoncompliant(HttpServletRequest request, java.sql.Connection connection) {
+        final String favoriteColor = request.getParameter("favoriteColor");
+        try {
+            String sql = "SELECT * FROM people WHERE favorite_color='" + favoriteColor + "'";
+            java.sql.Statement statement = connection.createStatement();
+            // Noncompliant: user-given input is not sanitized before use.
+            statement.execute(sql);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected an SQL command that might use unsanitized input. This can result in an SQL injection. To increase the security of your code, sanitize inputs before using them to form a query string.

[Learn more](https://cwe.mitre.org/data/definitions/89.html)